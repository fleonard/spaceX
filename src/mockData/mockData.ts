const dragon = {
  id: "dragon1",
  name: "Dragon 1",
  type: "capsule",
  active: true,
  crew_capacity: 0,
  sidewall_angle_deg: 15,
  orbit_duration_yr: 2,
  dry_mass_kg: 4200,
  dry_mass_lb: 9300,
  first_flight: "2010-12-8",
  flickr_images: [
    "https://imgur.com/DaCfMsj.jpg",
    "https://imgur.com/azYafd8.jpg",
  ],
  heat_shield: {
    material: "PICA-X",
    size_meters: 3.6,
    temp_degrees: 3000,
    dev_partner: "NASA",
  },
  thrusters: [
    {
      type: "Draco",
      amount: 18,
      pods: 4,
      isp: 300,
      fuel_1: "nitrogen tetroxide",
      fuel_2: "monomethylhydrazine",
      thrust: {
        kN: 0.4,
        lbf: 90,
      },
    },
  ],
  launch_payload_mass: {
    kg: 6000,
    lb: 13228,
  },
  launch_payload_vol: {
    cubic_meters: 25,
    cubic_feet: 883,
  },
  return_payload_mass: {
    kg: 3000,
    lb: 6614,
  },
  return_payload_vol: {
    cubic_meters: 11,
    cubic_feet: 388,
  },
  pressurized_capsule: {
    payload_volume: {
      cubic_meters: 11,
      cubic_feet: 388,
    },
  },
  trunk: {
    trunk_volume: {
      cubic_meters: 14,
      cubic_feet: 494,
    },
    cargo: {
      solar_array: 2,
      unpressurized_cargo: true,
    },
  },
  height_w_trunk: {
    meters: 7.2,
    feet: 23.6,
  },
  diameter: {
    meters: 3.7,
    feet: 12,
  },
  wikipedia: "https://en.wikipedia.org/wiki/SpaceX_Dragon",
  description:
    "Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS).",
};

const rocket = {
  id: 1,
  active: false,
  stages: 2,
  boosters: 0,
  cost_per_launch: 6700000,
  success_rate_pct: 40,
  first_flight: "2006-03-24",
  country: "Republic of the Marshall Islands",
  company: "SpaceX",
  height: {
    meters: 22.25,
    feet: 73,
  },
  diameter: {
    meters: 1.68,
    feet: 5.5,
  },
  mass: {
    kg: 30146,
    lb: 66460,
  },
  payload_weights: [
    {
      id: "leo",
      name: "Low Earth Orbit",
      kg: 450,
      lb: 992,
    },
  ],
  first_stage: {
    reusable: false,
    engines: 1,
    fuel_amount_tons: 44.3,
    burn_time_sec: 169,
    thrust_sea_level: {
      kN: 420,
      lbf: 94000,
    },
    thrust_vacuum: {
      kN: 480,
      lbf: 110000,
    },
  },
  second_stage: {
    reusable: false,
    engines: 1,
    fuel_amount_tons: 3.38,
    burn_time_sec: 378,
    thrust: {
      kN: 31,
      lbf: 7000,
    },
    payloads: {
      option_1: "composite fairing",
      composite_fairing: {
        height: {
          meters: 3.5,
          feet: 11.5,
        },
        diameter: {
          meters: 1.5,
          feet: 4.9,
        },
      },
    },
  },
  engines: {
    number: 1,
    type: "merlin",
    version: "1C",
    layout: "single",
    engine_loss_max: 0,
    propellant_1: "liquid oxygen",
    propellant_2: "RP-1 kerosene",
    thrust_sea_level: {
      kN: 420,
      lbf: 94000,
    },
    thrust_vacuum: {
      kN: 480,
      lbf: 110000,
    },
    thrust_to_weight: 96,
  },
  landing_legs: {
    number: 0,
    material: null,
  },
  wikipedia: "https://en.wikipedia.org/wiki/Falcon_1",
  description:
    "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
  rocket_id: "falcon1",
  rocket_name: "Falcon 1",
  rocket_type: "rocket",
  flickr_images: [
    "https://imgur.com/DaCfMsj.jpg",
    "https://imgur.com/azYafd8.jpg",
  ],
};

export { rocket, dragon };
