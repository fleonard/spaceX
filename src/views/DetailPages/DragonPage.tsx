import React from "react";
import { Redirect, useLocation } from "react-router-dom";

import { TableBody, TableCell, TableRow } from "@material-ui/core";

import { Dragon } from "../../types/apiTypes";

import { ColumnContainer } from "../../globalStyles";

import {
  StyledSection,
  ImageContainer,
  InfoContainer,
  StyledTable,
} from "./StyledDetailPages";

interface LocationState {
  item: Dragon;
}

const DragonPage: React.FC = () => {
  const location = useLocation();

  if (location.state === undefined) {
    return <Redirect to="/dragons" />;
  }

  const dragon = (location.state as LocationState).item;

  const {
    crew_capacity,
    description,
    diameter,
    dry_mass_kg,
    dry_mass_lb,
    first_flight,
    flickr_images,
    heat_shield,
    height_w_trunk,
    launch_payload_mass,
    launch_payload_vol,
    name,
    return_payload_mass,
    return_payload_vol,
    thrusters,
    trunk,
    type,
    wikipedia,
  } = dragon;

  return (
    <ColumnContainer>
      <StyledSection data-testid="informations">
        <ImageContainer>
          <img src={flickr_images[0]} alt={name} />
        </ImageContainer>
        <InfoContainer>
          <h2>{name}</h2>
          <p>{description}</p>
          <h3>Additional Informations</h3>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Crew capacity
                </TableCell>
                <TableCell align="right">{crew_capacity}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Frist flight
                </TableCell>
                <TableCell align="right">{first_flight}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Type
                </TableCell>
                <TableCell align="right">{type}</TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
          <a href={wikipedia} target="_blank" rel="noopener noreferrer">
            Checkout wikipidia page for more.
          </a>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="overview">
        <InfoContainer>
          <h3>Dragon</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Diameter
                </TableCell>
                <TableCell align="right">
                  {diameter.meters} m<span> / {diameter.feet} ft</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Height with trunk
                </TableCell>
                <TableCell align="right">
                  {height_w_trunk.meters} m
                  <span> / {height_w_trunk.feet} ft</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Dry Mass
                </TableCell>
                <TableCell align="right">
                  {dry_mass_kg} kg<span> / {dry_mass_lb} lb</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Launch payload mass
                </TableCell>
                <TableCell align="right">
                  {launch_payload_mass.kg} kg
                  <span> / {launch_payload_mass.lb} lb</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Launch payload volume
                </TableCell>
                <TableCell align="right">
                  {launch_payload_vol.cubic_meters} m³
                  <span> / {launch_payload_vol.cubic_feet} ft³</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Return payload mass
                </TableCell>
                <TableCell align="right">
                  {return_payload_mass.kg} kg
                  <span> / {return_payload_mass.lb} lb</span>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Return payload volume
                </TableCell>
                <TableCell align="right">
                  {return_payload_vol.cubic_meters} m³
                  <span> / {return_payload_vol.cubic_feet} ft³</span>
                </TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
        {flickr_images[1].length ? (
          <ImageContainer>
            <img src={flickr_images[1]} alt={name} />
          </ImageContainer>
        ) : (
          ""
        )}
      </StyledSection>
      <StyledSection data-testid="trunk">
        <InfoContainer>
          <h3>Trunk</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  {trunk.cargo.unpressurized_cargo ? "Unpressurized " : ""}Cargo
                </TableCell>
                <TableCell align="right">
                  Solar array: {trunk.cargo.solar_array}
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Volume
                </TableCell>
                <TableCell align="right">
                  {trunk.trunk_volume.cubic_meters} m³
                  <span> / {trunk.trunk_volume.cubic_feet} ft³</span>
                </TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="heat-shield">
        <InfoContainer>
          <h3>Heat Shield</h3>
          <h2>Overview</h2>
          <StyledTable aria-label="simple table">
            <TableBody>
              <TableRow>
                <TableCell component="th" scope="row">
                  Dev Partner
                </TableCell>
                <TableCell align="right">{heat_shield.dev_partner}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Material
                </TableCell>
                <TableCell align="right">{heat_shield.material}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Size
                </TableCell>
                <TableCell align="right">{heat_shield.size_meters} m</TableCell>
              </TableRow>
              <TableRow>
                <TableCell component="th" scope="row">
                  Temperature
                </TableCell>
                <TableCell align="right">{heat_shield.temp_degrees}°</TableCell>
              </TableRow>
            </TableBody>
          </StyledTable>
        </InfoContainer>
      </StyledSection>
      <StyledSection data-testid="thrusters">
        <InfoContainer>
          <h3>Thrusters</h3>
          <h2>Overview</h2>
          {thrusters.map((t) => (
            <div key={`id_${t.type}`}>
              <h4>{t.type}</h4>
              <StyledTable aria-label="simple table">
                <TableBody>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      Pods
                    </TableCell>
                    <TableCell align="right">{t.pods}</TableCell>
                  </TableRow>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      isp
                    </TableCell>
                    <TableCell align="right">{t.isp}</TableCell>
                  </TableRow>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      Amount
                    </TableCell>
                    <TableCell align="right">{t.amount}</TableCell>
                  </TableRow>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      Fuel 1
                    </TableCell>
                    <TableCell align="right">{t.fuel_1}</TableCell>
                  </TableRow>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      Fuel 2
                    </TableCell>
                    <TableCell align="right">{t.fuel_2}</TableCell>
                  </TableRow>
                  <TableRow>
                    <TableCell component="th" scope="row">
                      Thrust
                    </TableCell>
                    <TableCell align="right">
                      {t.thrust.kN} kN
                      <span> / {t.thrust.lbf} lbf</span>
                    </TableCell>
                  </TableRow>
                </TableBody>
              </StyledTable>
            </div>
          ))}
        </InfoContainer>
      </StyledSection>
    </ColumnContainer>
  );
};

export default DragonPage;
