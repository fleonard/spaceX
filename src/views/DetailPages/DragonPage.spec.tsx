import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import "jest-styled-components";

import { dragon } from "../../mockData/mockData";

import DragonPage from "./DragonPage";

describe("DragonPage Component", () => {
  it("renders correctly", () => {
    const history = createMemoryHistory();
    const state = { item: dragon };
    history.push("/dragons/dragon1", state);

    const { container, getByText, getByTestId } = render(
      <Router history={history}>
        <DragonPage />
      </Router>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText("Dragon 1")).toBeDefined();
    expect(getByTestId("informations")).toBeDefined();
    expect(getByTestId("overview")).toBeDefined();
    expect(getByTestId("trunk")).toBeDefined();
    expect(getByTestId("heat-shield")).toBeDefined();
    expect(getByTestId("thrusters")).toBeDefined();
  });
});
