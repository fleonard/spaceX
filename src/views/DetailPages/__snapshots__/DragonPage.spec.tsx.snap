// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragonPage Component renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px 20px;
}

.c1 {
  border-bottom: 1px solid #dadce0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 25px 0;
}

.c1:first-child {
  padding-top: 25px;
}

.c1:last-child {
  border: 0;
}

.c2 {
  display: block;
  padding-bottom: 20px;
  text-align: center;
  width: 100%;
}

.c2 img {
  height: auto;
  max-width: 600px;
  width: 100%;
}

.c3 {
  padding-bottom: 20px;
  width: 100%;
}

.c3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 table span {
  color: #868686;
}

.c4.c4 {
  margin-bottom: 20px;
}

@media (min-width:1000px) {
  .c1 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 80px 0;
  }
}

@media (min-width:1000px) {
  .c2 {
    padding-bottom: 0;
    padding-right: 20px;
  }
}

@media (min-width:1000px) {
  .c3 {
    padding-bottom: 0px;
    padding-right: 20px;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
    data-testid="informations"
  >
    <div
      class="c2"
    >
      <img
        alt="Dragon 1"
        src="https://imgur.com/DaCfMsj.jpg"
      />
    </div>
    <div
      class="c3"
    >
      <h2>
        Dragon 1
      </h2>
      <p>
        Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California. Dragon is launched into space by the SpaceX Falcon 9 two-stage-to-orbit launch vehicle. The Dragon spacecraft was originally designed for human travel, but so far has only been used to deliver cargo to the International Space Station (ISS).
      </p>
      <h3>
        Additional Informations
      </h3>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Crew capacity
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              0
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Frist flight
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2010-12-8
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Type
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              capsule
            </td>
          </tr>
        </tbody>
      </table>
      <a
        href="https://en.wikipedia.org/wiki/SpaceX_Dragon"
        rel="noopener noreferrer"
        target="_blank"
      >
        Checkout wikipidia page for more.
      </a>
    </div>
  </section>
  <section
    class="c1"
    data-testid="overview"
  >
    <div
      class="c3"
    >
      <h3>
        Dragon
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Diameter
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              3.7
               m
              <span>
                 / 
                12
                 ft
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Height with trunk
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              7.2
               m
              <span>
                 / 
                23.6
                 ft
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Dry Mass
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              4200
               kg
              <span>
                 / 
                9300
                 lb
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Launch payload mass
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              6000
               kg
              <span>
                 / 
                13228
                 lb
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Launch payload volume
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              25
               m³
              <span>
                 / 
                883
                 ft³
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Return payload mass
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              3000
               kg
              <span>
                 / 
                6614
                 lb
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Return payload volume
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              11
               m³
              <span>
                 / 
                388
                 ft³
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="c2"
    >
      <img
        alt="Dragon 1"
        src="https://imgur.com/azYafd8.jpg"
      />
    </div>
  </section>
  <section
    class="c1"
    data-testid="trunk"
  >
    <div
      class="c3"
    >
      <h3>
        Trunk
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Unpressurized 
              Cargo
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              Solar array: 
              2
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Volume
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              14
               m³
              <span>
                 / 
                494
                 ft³
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
  <section
    class="c1"
    data-testid="heat-shield"
  >
    <div
      class="c3"
    >
      <h3>
        Heat Shield
      </h3>
      <h2>
        Overview
      </h2>
      <table
        aria-label="simple table"
        class="MuiTable-root c4"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Dev Partner
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              NASA
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Material
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              PICA-X
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Size
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              3.6
               m
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Temperature
            </th>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              3000
              °
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
  <section
    class="c1"
    data-testid="thrusters"
  >
    <div
      class="c3"
    >
      <h3>
        Thrusters
      </h3>
      <h2>
        Overview
      </h2>
      <div>
        <h4>
          Draco
        </h4>
        <table
          aria-label="simple table"
          class="MuiTable-root c4"
        >
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                Pods
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                4
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                isp
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                300
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                Amount
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                18
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                Fuel 1
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                nitrogen tetroxide
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                Fuel 2
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                monomethylhydrazine
              </td>
            </tr>
            <tr
              class="MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body"
                role="cell"
                scope="row"
              >
                Thrust
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                0.4
                 kN
                <span>
                   / 
                  90
                   lbf
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </section>
</div>
`;
